### PR TITLE
[CARBONDATA-3269] Fix ArrayIndexOutOfBoundsException of Range_Column when using KryoSerializer

### PR DESCRIPTION
**Pull request by QiangCai**
Fix ArrayIndexOutOfBoundsException of Range_Column when using KryoSerializer

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2437/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10695/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2665/

**Comment by ravipesala** [comment]
LGTM

File: integration/spark-common/src/main/scala/org/apache/spark/DataSkewRangePartitioner.scala [hunk 1]
```diff
@@ -202,7 +202,7 @@ class DataSkewRangePartitioner[K: Ordering : ClassTag, V](
     }
   }
 
-  private val skewPartitions: Int = if (skewCount == 0) {
+  private var skewPartitions: Int = if (skewCount == 0) {
```
Comment (xuchuanyin) [inline integration/spark-common/src/main/scala/org/apache/spark/DataSkewRangePartitioner.scala]: What does skewPartitions and skewCount mean respectively?

Comment (QiangCai) [reply integration/spark-common/src/main/scala/org/apache/spark/DataSkewRangePartitioner.scala]: shewCount: the count of shew index
skePartitions: the number of skew partitions

**Pull request closed without merging**